service/worker/scanner/tasklist/handler.go

```diff
@@ -47,60 +48,45 @@ const scannerTaskListPrefix = "cadence-sys-tl-scanner"
 // with the assumption that the executor will schedule this task later
 //
 // Each loop of the handler proceeds as follows
-//    - Retrieve the next batch of tasks sorted by task_id for this task-list from persistence
+//    - Attempt to delete tasks up to the persisted ACK level.
 //    - If there are 0 tasks for this task-list, try deleting the task-list if its idle
-//    - If any of the tasks in the batch isn't expired, we are done. Since tasks are retrieved
-//      in sorted order, if one of the tasks isn't expired, chances are, none of the tasks above
-//      it are expired as well - so, we give up and wait for the next run
-//    - Delete the entire batch of tasks
 //    - If the number of tasks retrieved is less than batchSize, there are no more tasks in the task-list
```

Comment:
this comment is quite hard to understand, maybe too misleading. Maybe say :
`If the number of tasks from completeTasks is less than the batch size, ...`